cpp/src/io/json/experimental/read_json.cu

```diff
@@ -66,9 +80,27 @@ rmm::device_uvector<char> ingest_raw_input(host_span<std::unique_ptr<datasource>
             destination, h_buffer->data(), h_buffer->size(), cudaMemcpyDefault, stream.value()));
           bytes_read += h_buffer->size();
         }
+        delimiter_map.push_back(bytes_read);
+        bytes_read += num_delimiter_chars;
       }
     }
 
+    // If this is a multi-file source, we scatter the JSON line delimiters between files
+    if (sources.size() > 1) {
+      static_assert(num_delimiter_chars == 1,
+                    "Currently only single-character delimiters are supported");
+      auto const delimiter_source = thrust::make_constant_iterator('\n');
```

Comment:
I think we need to also look for `\r\n` in addition to `\n`

Reply:
Thanks, this shouldn't be an issue here. We just need to make sure that the last line of file `i` and the first line of file `i+1` are parsed as different JSON lines. Inserting a `\n` is sufficient for this.